addon/components/frost-list.js

```diff
@@ -160,16 +167,16 @@ export default Component.extend({
 
     _select ({isRangeSelect, isSpecificSelect, item}) {
       const items = this.get('items')
-      const clonedSelectedItems = this.get('selectedItems').slice()
+      const clonedSelectedItems = A(this.get('selectedItems').slice())
       const _rangeState = this.get('_rangeState')
 
       // Selects are proccessed in order of precedence: specific, range, basic
       if (isSpecificSelect) {
-        selection.specific(clonedSelectedItems, item, _rangeState)
+        selection.specific(clonedSelectedItems, item, _rangeState, this.get('itemComparator'))
```

Comment:
Future note, best to store this against a `const` and then refer for each usage (when more than one usage is present)